receiver/awscontainerinsightreceiver/internal/host/utils.go

```diff
@@ -0,0 +1,22 @@
+package host
+
+import "time"
+
+// execute the refresh() function periodically with the given refresh interval
+// until shouldRefresh() return false or the shutdownC is closed
+func refreshUntil(refresh func(), refreshInterval time.Duration,
```

Comment:
This should probably take a context and select on `<-ctx.Done()` rather than a stop channel.  Accepting a context allows it to pass that context down to the `refresh()` invocation so that it can be used in there to stop any work once the context is cancelled.  This allows, for instance, stopping and cleaning up after in-flight network requests, if the relevant `*WithContext()` AWS SDK methods are used.